tests/unit/mixins/createStatefulChildrenMixin.ts

```diff
@@ -349,5 +356,102 @@ registerSuite({
 		}).then(() => {
 			assert.isTrue(List([ widget3 ]).equals(parent.children));
 		});
+	},
+	'createChild()': {
+		'creation during mixin'() {
+			let p: Promise<[string, Renderable]>;
+			const registry = Object.create(widgetRegistry);
+			const createFoo = compose({})
+				.mixin({
+					mixin: createStatefulChildrenMixin,
+					initialize(instance) {
+						p = instance.createChild(createRenderable, <RenderableOptions> {
+							render() {
+								return h('div');
+							}
+						});
+					}
+				});
+			const foo = createFoo({
+				registryProvider: {
+					get(type: string) {
+						return type === 'widgets' ? registry : null;
+					}
+				}
+			});
+			assert(foo);
+			assert(p);
+			return p.then((result) => {
+				const [ id, widget ] = result;
+				assert.include(id, 'undefined-child');
```

Comment:
Do we want to generate an `id` that contains `undefined`? Does this mean that there could be `id` collisions between widgets where the parent does not contain `id`?